sources/mysql/streaming/ddl.go

```
@@ -0,0 +1,165 @@
+package streaming
+
+import (
+	"fmt"
+	"log/slog"
+	"slices"
+	"time"
+
+	"github.com/go-mysql-org/go-mysql/replication"
+
+	"github.com/artie-labs/reader/lib/antlr"
+)
+
+type Column struct {
+	Name       string
+	DataType   string
+	PrimaryKey bool
+}
+
+type TableAdapter struct {
+	columns []Column
+}
+
+type SchemaAdapter struct {
+	adapters map[string]TableAdapter
+}
+
+func (i *Iterator) persistAndProcessDDL(evt *replication.QueryEvent, ts time.Time) error {
+	if evt.ErrorCode != 0 {
+		// Don't process a non-zero error code DDL.
+		return nil
+	}
+
+	query := string(evt.Query)
+	schemaHistory := SchemaHistory{
+		Query: query,
+		Ts:    ts,
+	}
+
+	if err := i.schemaHistoryList.Push(schemaHistory); err != nil {
+		return fmt.Errorf("failed to push schema history: %w", err)
+	}
+
+	return i.schemaAdapter.ApplyDDL(query)
+}
+
+func (s *SchemaAdapter) ApplyDDL(query string) error {
+	results, err := antlr.Parse(query)
+	if err != nil {
+		if antlr.IsParseError(err) {
+			return nil
```

We're swallowing the errors here?

---

This is the error: https://github.com/artie-labs/reader/blob/19f0a9402009c881e1e3cb6053824e1472397e8e/lib/antlr/parse.go#L56

---

It's not really an error, it's more of the fact that we don't support this kind of DDL.

---

If we don't support the DDL shouldn't we fail rather than possibly miss a DDL update?

---

The way I built the DDL Parser is that we are already parsing all the relevant DDL updates. The irrelevant ones have an error being thrown for now.

I can add a warning to it though.

---

If we wanted to deliberately fail, we would need to comb through all the possible DDL permutations which is not done right now

---

Hm, thinking more about this - it might be better to explicitly list out the event types that we don't care about rather than use a fallback like this.

I was thinking this way would be safer for Production since it won't just crash and fail. However, if we did miss a DDL event, it would mask the error and make it harder to troubleshoot

---

Thanks for the flag here, just redid it to list out all the event types.

I suspect we'll need to do more later, but all is well